arboretum/ibm_cloud/fetchers/fetch_cluster_resource.py

```diff
@@ -0,0 +1,151 @@
+            'org.ibm_cloud.cluster_resources.types', RESOURCE_TYPES_DEFAULT
+        )
+        cls.tempdir = tempfile.TemporaryDirectory()
+        return cls
+
+    @classmethod
+    def tearDownClass(cls):
+        """Cleanup class."""
+        cls.tempdir.cleanup()
+
+    @store_raw_evidence('ibm_cloud/cluster_resources.json')
+    def fetch_cluster_resource(self):
+        """Fetch cluster resources."""
+        cluster_list_evidence = get_evidence_dependency(
+            'raw/ibm_cloud/cluster_list.json', self.locker
+        )
+        cluster_list = cluster_list_evidence.content_as_json
+        resources = {}
+        for account in cluster_list:
+            api_key = self.config.creds.get('ibm_cloud', f'{account}_api_key')
+            access_token, refresh_token = get_tokens(api_key)
+            headers = {
+                'Accept': 'application/json',
+                'Authorization': f'Bearer {access_token}',
+                'X-Auth-Refresh-Token': refresh_token
+            }
+            self.session('https://containers.cloud.ibm.com', **headers)
+
+            resources[account] = []
+            for cluster in cluster_list[account]:
+                self.session('https://containers.cloud.ibm.com', **headers)
+                config_url = f'/global/v1/clusters/{cluster["id"]}/config'
+                resp = self.session().get(config_url)
+                resp.raise_for_status()
+                cluster_config = zipfile.ZipFile(io.BytesIO(resp.content))
+                if cluster['type'] == 'kubernetes':
+                    cluster_token, ca_cert = self._get_iks_credentials(
+                        cluster_config
+                    )
+                elif cluster['type'] == 'openshift':
+                    cluster_token = self._get_roks_credentials(
+                        cluster, api_key
+                    )
+                    ca_cert = None
+                self.session(cluster['serverURL'], **headers)
+                cluster['resources'] = get_cluster_resources(
+                    self.session(),
+                    cluster_token,
+                    self.resource_types,
+                    ca_cert
+                )
+                resources[account].append(cluster)
+
+        return json.dumps(resources)
+
+    def _get_iks_credentials(self, cluster_config):
+        """Get credentials for an IKS cluster.
+
+        This function implements the procedure described in
+        https://cloud.ibm.com/apidocs/kubernetes#getclusterconfig
+        """
+        for name in cluster_config.namelist():
+            p = pathlib.Path(name)
+            if p.name.startswith('kube-config'):
+                kubeconfig = yaml.safe_load(cluster_config.read(name))
+                usr = kubeconfig['users'][0]['user']
+                cluster_token = usr['auth-provider']['config']['id-token']
+            if p.name.endswith('.pem'):
+                t = pathlib.Path(self.tempdir.name)
```

Comment:
What's the point of doing this?  I believe that this just turns the string represented by `self.tempdir.name` into a `Path` object.  Does the `extract` method on the cluster_config ZipFile object accept a `Path` object?  The docs seem to imply that the `path` argument is meant to be a string.  If it's meant to be a string then I suggest you do away with this unneeded wrapping of the path string as a Path object.

Reply:
Test results showed me that `Path` is also acceptable, but as you mentioned documentation assumes this string.  I changed as you suggested in the comment below.